### PR TITLE
docs: Clarify commit message guidelines

### PR DESCRIPTION
**Pull request by merland**
In this small PR, the gist of [this helpful and informative comment from @fanquake](https://github.com/bitcoin/bitcoin/pull/14583#issuecomment-433668211) is added to the official contributing instructions, to help future first-time contributors get their commit messages right.

### CODE REVIEW

**Comment by practicalswift** [comment]
ACK 73f358401cd452577a516ca565ff8f6742245808

**Comment by ch4ot1c** [comment]
Helpful, thank you!

**Comment by jnewbery** [comment]
"any @ mentions should be removed" isn't specific to the squashing commits section. There shouldn't be @ mentions in any commit messages at all.

**Comment by merland** [comment]
@jnewbery Thanks. Do you know if this is mentioned anywhere in the bitcoin repo?

**Comment by jnewbery** [comment]
I don't know. It should be in CONTRIBUTING.md if it's anywhere.

**Comment by merland** [comment]
@jnewbery I updated the commit after your suggestion, please have another look.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -135,6 +137,9 @@ before it will be merged. The basic squashing workflow is shown below.
     # Save and quit.
     git push -f # (force push to GitHub)
 
+The commit message should be updated so that it doesn't include all the
+messages from the interim commits.
```
Comment (hebasto) [inline CONTRIBUTING.md]: IMO, this requirement is not suitable for all cases.

Comment (merland) [reply CONTRIBUTING.md]: Interesting! Would you (or someone else) care to elaborate? 
Personally I don't really know, but it seems like something that would benefit from a set guideline.

Comment (hebasto) [reply CONTRIBUTING.md]: e.g., 869193f5a6398bb9e4a6f4a7aeb8d2203193c80f
That commit was squashed from two ones. And its message _does_ include all the messages from the squashed commits.

Comment (merland) [reply CONTRIBUTING.md]: I think I see what you are saying; there can be cases where just listing the interim commits happens to form a coherent message. 

New suggestion:
`Please update the resulting commit message if needed, it should read as a coherent message. In most cases this means that you should not just list the interim commits.`

Comment (hebasto) [reply CONTRIBUTING.md]: LGTM

**Comment by merland** [comment]
Commit updated after comments

**Comment by ismail120572** [comment]
Mohon maaf tentu akan ditindaklanjuti
Terima kasih

Ismail

Pada tanggal Sel, 30 Okt 2018 14:05, Martin Erlandsson <
notifications@github.com> menulis:

**Comment by hebasto** [comment]
ACK 0e6de3aacb8ebbf2617e8c11b8dae61acdd79816

**Comment by fanquake** [comment]
utACK 0e6de3a

**Pull request merged**